### PR TITLE
`azurerm_iotcentral_application` - fix default `display_name`

### PR DESCRIPTION
**Pull request by myc2h6o**
In the document and Update function, `display_name` is defaults to the ResourceName. However, in Create function it's ResourceGroupName, fixing the value there.

https://github.com/hashicorp/terraform-provider-azurerm/blob/a5a6c08e1c03b722a4b271d19d581bceb01433e6/website/docs/r/iotcentral_application.html.markdown?plain=1#L49

https://github.com/hashicorp/terraform-provider-azurerm/blob/a5a6c08e1c03b722a4b271d19d581bceb01433e6/internal/services/iotcentral/iotcentral_application_resource.go#L190-L193

### CODE REVIEW

**Comment by myc2h6o** [comment]
Test result
image

File: internal/services/iotcentral/iotcentral_application_resource.go [hunk 1]
```diff
@@ -134,7 +134,7 @@ func resourceIotCentralAppCreate(d *pluginsdk.ResourceData, meta interface{}) er
 
 	displayName := d.Get("display_name").(string)
 	if displayName == "" {
-		displayName = id.ResourceGroupName
+		displayName = id.ResourceName
```
Comment (katbyte) [inline internal/services/iotcentral/iotcentral_application_resource.go:137]: Given we are changing the existing default behaviour can we make this a 4.0 change and wrap it in the flag?

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by myc2h6o** [comment]
@katbyte thanks for reviewing the pr! I've moved the change behind the 4.0 flag and updated the corresponding document. Please take a look.

**Comment by myc2h6o** [comment]
Updated test result
image

**Review by katbyte — APPROVED** [review]
Thanks LGTM 🎧

**Pull request merged**